### ISSUE TITLE
StoreFileListener.onReleased() not called when Tailer read all events in the queue

### ISSUE THREAD

**Issue by keitaf** [OpenHFT/Chronicle-Queue#724]
It looks like the ChronicleQueue version `5.20.41` no longer calls `StoreFileListener.onReleased()` once a tailer is done reading all events in the queue.

Here is a code snippet to reproduce it.

```
        var tmpDir = Files.createTempDirectory("cq");

        var writeQueue = SingleChronicleQueueBuilder
            .binary(tmpDir)
            .rollCycle(RollCycles.MINUTELY)
            .build();

        for (int i = 0; i < 10; i++) {
            new Thread(() -> {
                var appender = writeQueue.acquireAppender();
                while (true) {
                    appender.writeText("text");
                    try {
                        Thread.sleep(100);
                    } catch (InterruptedException e) {}
                }
            }, "appender-" + i).start();
        }

        var readQueue = SingleChronicleQueueBuilder
            .binary(tmpDir)
            .readOnly(true)
            .rollCycle(RollCycles.MINUTELY)
            .storeFileListener((cycle, file) -> {
                System.out.println(file + " is released in " + Thread.currentThread().getName());
            })
            .build();

        new Thread(() -> {
            var tailer = readQueue.createTailer();
            while (true) {
                tailer.readText();
            }
        }, "tailer").start();
```

I can see messages like "{.cq4 file } is released in tailer" if running with `5.17.25`, but I don't see any message if running with `5.20.41`.

**Comment by JerryShea** [comment]
hi @keitaf is this the same as #694 ?

**Comment by peter-lawrey** [comment]
While the Tailer still exists, it holds onto the file until you either read the next cycle so it no longer needs it or you close it.
You don't appear to be doing either in this test.

**Comment by peter-lawrey** [comment]
Due to a refactoring issue, only appender file releases were being notified.

**Comment by keitaf** [comment]
@peter-lawrey thank you for fixing it quickly!

**Comment by keitaf** [comment]
@JerryShea yes looks like it's the same issue.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.20.42](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.20.42), [BOM-2.20.67](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.67)
